src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.AliasesAndUsings.cs

```diff
@@ -198,27 +221,175 @@ public ImmutableArray<NamespaceOrTypeAndUsingDirective> GetUsingNamespacesOrType
                     throw ExceptionUtilities.UnexpectedValue(declarationSyntax);
             }
 
-            foreach (var declaration in _mergedDeclaration.Declarations)
+            return GetAliasesAndUsings(declarationSyntax).GetUsingNamespacesOrTypes(this, declarationSyntax, basesBeingResolved);
+        }
+
+        private Imports GetGlobalUsingImports(ConsList<TypeSymbol>? basesBeingResolved)
+        {
+            return GetMergedGlobalAliasesAndUsings(basesBeingResolved).Imports;
+        }
+
+        private ImmutableDictionary<string, AliasAndUsingDirective> GetGlobalUsingAliasesMap(ConsList<TypeSymbol>? basesBeingResolved)
+        {
+            return GetMergedGlobalAliasesAndUsings(basesBeingResolved).UsingAliasesMap!;
```

Comment:
>! [](start = 86, length = 1)

nit: should use an assertion for nullability check instead

Reply:
We have an explicit nullability guideline for this. `!` should be used for things that are checked elsewhere in the method.
[https://github.com/dotnet/roslyn/blob/main/docs/contributing/Nullable%20Annotations.md](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Nullable%20Annotations.md)

---
In reply to: [595733537](https://github.com/dotnet/roslyn/pull/51855#discussion_r595733537) [](ancestors = 595733537)

Reply:
The private helper that is used to get the value guarantees not null value. I see no benefit in asserting that especially that consumer will crash immediately if a null value is returned.

---
In reply to: [596253024](https://github.com/dotnet/roslyn/pull/51855#discussion_r596253024) [](ancestors = 596253024,595733537)